docs/manual/src/Upgrading.md

```diff
@@ -0,0 +1,35 @@
+# v0.28.x -> v0.29.x
+
+## Custom types
+
+Custom types are now implemented using a macro, rather than implementing the
+`UniffiCustomTypeConverter` trait.
+
+Before:
+
+```rust
+impl UniffiCustomTypeConverter for MyType {
+    type Builtin = UniFFIBuiltinType;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        ...
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        ...
+    }
+}
+```
+
+After:
+
+```
+uniffi::custom_type!(MyType, UniFFIBuiltinType, {
+    try_from_existing: |val| { ... },
```

Comment:
really nit-picking here, but I wonder if this line should say, eg, `try_from_existing: |val| { Ok(...) },` to highlight the fact this is a `Result<>`?